Project/zepeto-modules/Assets/Zepeto Gesture Module/ZepetoScript/GestureLoader.ts

```diff
@@ -51,24 +55,66 @@ export default class GestureLoader extends ZepetoScriptBehaviour {
         newThumb.GetComponent<Thumbnail>().content = content;
 
         // Button Listener for each thumbnail
-        newThumb.GetComponent<Button>().onClick.AddListener(() => {
+        newThumb.GetComponent<Button>().onClick.AddListener(() => {  
             this.LoadAnimation(content);
         });
-        
         this.thumbnails.push(newThumb);
     }
 
     // 3. Loading Animation
-    private LoadAnimation(content: Content) {
+    private LoadAnimation(content: Content ) {
         // Verify animation load
         if (!content.IsDownloadedAnimation) {
             // If the animation has not been downloaded, download it.
             content.DownloadAnimation(() => {
                 // play animation clip
-                this._myCharacter.SetGesture(content.AnimationClip);
-            });
+                this.runAnimation(content.AnimationClip, content.Keywords)
+            });    
         } else {
-            this._myCharacter.SetGesture(content.AnimationClip);
+            this.runAnimation(content.AnimationClip, content.Keywords)
+        }
+    }
+       
+    // A function to run an animation, 
+    private runAnimation(animation: AnimationClip, type: OfficialContentType[] )
```

Comment:
I thought it would be better to not use the whole `content` class and only use the specific elements of the class that we need. In this case, `AnimationClip` and the `Keywords` as it is less expensive to do so.

Reply:
OK sounds good.